src/main/groovy/org/ajoberstar/grgit/auth/AuthConfig.java

```diff
@@ -0,0 +1,266 @@
+
+/**
+ * Stores configuration options for how to authenticate with remote repositories.
+ *
+ * <p>The following system properties can be set to configure how authentication is performed with
+ * remote repositories. All "allow" properties default to {@code true}.
+ *
+ * <ul>
+ *   <li>{@code org.ajoberstar.grgit.auth.force={hardcoded|interactive|sshagent|pageant}}
+ *   <li>{@code org.ajoberstar.grgit.auth.hardcoded.allow={true|false}}
+ *   <li>{@code org.ajoberstar.grgit.auth.interactive.allow={true|false}}
+ *   <li>{@code org.ajoberstar.grgit.auth.sshagent.allow={true|false}}
+ *   <li>{@code org.ajoberstar.grgit.auth.pageant.allow={true|false}}
+ * </ul>
+ *
+ * <p>In order to set default hardocded credentials, use the following properties. Note that unless
+ * hardcoded credentials are disabled, using these properties will supersede the use of interactive
+ * creds, ssh-agent, or Pageant. However, they will not take precedence over credentials provided
+ * directly to a repository during the clone, init, or open.
+ *
+ * <ul>
+ *   <li>{@code org.ajoberstar.grgit.auth.username=<username>}
+ *   <li>{@code org.ajoberstar.grgit.auth.password=<password>}
+ * </ul>
+ *
+ * <p>Hardcoded credentials can alternately be provided with environment variables. These take a
+ * lower precedence than the system properties, but all other considerations are the same.
+ *
+ * <ul>
+ *   <li>{@code GRGIT_USER=<username>}
+ *   <li>{@code GRGIT_PASS=<password>}
+ * </ul>
+ *
+ * <p>To customize SSH credentials use the following properties. In order to add a non-standard SSH
+ * key to use as your credentials, use the first following property. In case your private key is
+ * protected by a passphrase, use the second property.
+ *
+ * <ul>
+ *   <li>{@code org.ajoberstar.grgit.auth.ssh.private=<path.to.private.key>}
+ *   <li>{@code org.ajoberstar.grgit.auth.ssh.passphrase=<passphrase>}
+ * </ul>
+ *
+ * <p>In order to customize the JSch session config use a property of the following format (possible
+ * values <a
+ * href="http://epaul.github.io/jsch-documentation/javadoc/com/jcraft/jsch/JSch.html#setConfig(java.lang.String,
+ * java.lang.String)">JSch documentation</a>):
+ *
+ * <ul>
+ *   <li>{@code org.ajoberstar.grgit.auth.session.config.<key>=<value>}
+ * </ul>
+ *
+ * <p>The following order is used to determine which authentication option is used.
+ *
+ * <ol>
+ *   <li>Hardcoded credentials, if provided.
+ *   <li>Ssh-Agent, if available.
+ *   <li>Pageant, if available.
+ *   <li>Interactive credentials, if needed.
+ * </ol>
+ *
+ * @since 0.2.0
+ */
+public class AuthConfig {
+  /** System property name used to force a specific authentication option. */
+  public static final String FORCE_OPTION = "org.ajoberstar.grgit.auth.force";
+
+  public static final String USERNAME_OPTION = "org.ajoberstar.grgit.auth.username";
+  public static final String PASSWORD_OPTION = "org.ajoberstar.grgit.auth.password";
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this hard-coded password. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.ajoberstar.com/coding_rules#rule_key=squid%3AS2068)